### ISSUE TITLE
Can't sign in

### ISSUE THREAD

**Issue by Mortbertjh** [Metalnem/bandcamp-downloader#1]
I'm trying to use this tool but i'm having problems, specifically:

Unhandled exception: System.Exception: Max number of login attempts exceeded.
   at Api.Client.Login(String username, String password) in C:\Users\crisv\Downloads\Compressed\bandcamp-downloader-main\Api\Client.cs:line 129
   at Api.Client.GetCollection(Dictionary`2 queryString) in C:\Users\crisv\Downloads\Compressed\bandcamp-downloader-main\Api\Client.cs:line 99
   at Api.Client.GetCollection()+MoveNext() in C:\Users\crisv\Downloads\Compressed\bandcamp-downloader-main\Api\Client.cs:line 79
   at Api.Client.GetCollection()+System.Threading.Tasks.Sources.IValueTaskSource<System.Boolean>.GetResult()
   at BandcampDownloader.Program.List(String username, String password) in C:\Users\crisv\Downloads\Compressed\bandcamp-downloader-main\BandcampDownloader\Program.cs:line 40
   at BandcampDownloader.Program.List(String username, String password) in C:\Users\crisv\Downloads\Compressed\bandcamp-downloader-main\BandcampDownloader\Program.cs:line 40
   at System.CommandLine.Invocation.AnonymousCommandHandler.InvokeAsync(InvocationContext context)
   at System.CommandLine.Invocation.InvocationPipeline.<>c__DisplayClass4_0.<<BuildInvocationChain>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass17_0.<<UseParseErrorReporting>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass12_0.<<UseHelp>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass22_0.<<UseVersionOption>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass19_0.<<UseTypoCorrections>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c.<<UseSuggestDirective>b__18_0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass16_0.<<UseParseDirective>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c.<<RegisterWithDotnetSuggest>b__5_0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass8_0.<<UseExceptionHandler>b__0>d.MoveNext()

**Comment by Virusmater** [comment]
Same, after 2nd attempt to login with X-Bandcamp-Dm header, Bandcamp doesn't respond with X-Bandcamp-Pow

**Comment by Metalnem** [comment]
Fixed! Please try again and let me know how it goes.

**Comment by Virusmater** [comment]
Nice, I can log in now. Did Bandcamp change their login routine and you had to reverse engineer the new apk again?

**Comment by Metalnem** [comment]
There are three different algorithms to calculate X-Bandcamp-Dm header and they are based on different hash functions: SHA1, SHA256, and SHA512. When I first wrote this tool, Bandcamp servers were using the SHA256-based algorithm, so I reverse engineered only that specific algorithm. It appears that they recently switched back to SHA1, so I had to reverse that one, too. If they switch to SHA512, the tool will break again, but I can easily reverse engineer that one if needed.
